### PR TITLE
Update wording

### PR DESCRIPTION
**Pull request by runleonarun**
## What are you changing in this pull request and why?

Hey @jtcohen6 or @dbeatty10! @kbrock91 requested we change this a tiny bit. Can you let me know if this wording is OK? I also took the passive voice out to make the English a little simpler.

She also asked if we have a list / matrix available of which platforms support `constraints` and whether they are supported and/or enforced ? Or a link to a page with that info?

## Checklist

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

### CODE REVIEW

**Review by jtcohen6 — DISMISSED** [review]
@runleonarun This change in wording is good by me!

> She also asked if we have a list / matrix available of which platforms support `constraints` and whether they are supported and/or enforced ? Or a link to a page with that info?

Yes, great idea! I made a table of these a few months ago, to guide the implementation for the DWH adapters that we maintain: https://github.com/dbt-labs/dbt-core/issues/7067#issuecomment-1447281379.
- How do you think we should make that extensible, for all the other DWHs?
- Do you want to do that as part of the scope of this PR, or should we open a new issue?

**Comment by dbeatty10** [comment]
If we had adapter-specific feature support matrix pages similar to [this](https://docs.getdbt.com/reference/warehouse-setups/trino-setup), we could add a table for dbt model contract constraints.

Along the lines of https://github.com/dbt-labs/dbt-core/issues/7067#issuecomment-1447281379, here's some brainstorming on representing how different constraints will behave per adapter:

| constraint type | dbt-redshift              |
|-----------------|---------------------------|
| not_null        | 🌕 supported & enforced    |
| primary_key     | 🌗 supported; not enforced |
| foreign_key     | 🌗 supported; not enforced |
| unique          | 🌗 supported; not enforced |
| check           | 🌑 not supported           |

| constraint type |        dbt-redshift       |
|---------------|-------------------------|
| not_null        | 🌕 supported & enforced    |
| primary_key     | 🌓 supported; not enforced |
| foreign_key     | 🌓 supported; not enforced |
| unique          | 🌓 supported; not enforced |
| check           | 🌑 not supported           |

| constraint type | dbt-redshift              |
|-----------------|:---------------------------:|
| not_null        | 🌕    |
| primary_key     | 🌗 |
| foreign_key     | 🌗 |
| unique          | 🌗 |
| check           | 🌑           |

| constraint type | not_null               | primary_key               | foreign_key               | unique                    | check           |
|-----------------|------------------------|---------------------------|---------------------------|---------------------------|-----------------|
| dbt-redshift    | 🌕 supported & enforced | 🌗 supported; not enforced | 🌗 supported; not enforced | 🌗 supported; not enforced | 🌑 not supported |

| constraint type | not_null | primary_key | foreign_key | unique | check |
|-----------------|----------|-------------|-------------|--------|-------|
| dbt-redshift    | 🌕        | 🌗           | 🌗           | 🌗      | 🌑     |


| constraint type | not_null | primary_key | foreign_key | unique | check |
|-----------------|----------|-------------|-------------|--------|-------|
| dbt-redshift    | ✅        | 🟡           | 🟡           | 🟡      | ❌     |

| constraint type | not_null | primary_key | foreign_key | unique | check |
|-----------------|----------|-------------|-------------|--------|-------|
| dbt-redshift    | ✅        | 🟧           | 🟧           | 🟧      | ❌     |

| constraint type | not_null | primary_key | foreign_key | unique | check |
|-----------------|----------|-------------|-------------|--------|-------|
| dbt-redshift    | ✅        | ⚠️           | ⚠️           | ⚠️      | ❌     |

| constraint type | not_null | primary_key | foreign_key | unique | check |
|-----------------|----------|-------------|-------------|--------|-------|
| dbt-redshift    | ✅        | 🔄           | 🔄           | 🔄      | ❌     |

File: website/docs/docs/collaborate/govern/model-contracts.md [hunk 1]
```diff
@@ -23,9 +23,9 @@ While this is ideal for quick and iterative development, for some models, consta
 ## Where are contracts supported?
 
 At present, model contracts are supported for:
-- SQL models. Contracts are not yet supported for Python models.
-- Models materialized as `table`, `view`, and `incremental` (with `on_schema_change: append_new_columns`). Views offer limited support for column names and data types, but not `constraints`. Contracts are not supported for `ephemeral`-materialized models.
-- The most commonly used [data platforms](/docs/supported-data-platforms), including Snowflake, BigQuery, Databricks, and Redshift. However, the specific `constraints` that are supported or enforced can vary depending on the platform.
+- SQL models (not yet Python)
```
Comment (runleonarun) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
- SQL models. Contracts are not yet supported for Python models.
```

File: website/docs/docs/collaborate/govern/model-contracts.md [hunk 2]
```diff
@@ -24,8 +24,8 @@ While this is ideal for quick and iterative development, for some models, consta
 
 At present, model contracts are supported for:
 - SQL models. Contracts are not yet supported for Python models.
-- Models materialized as `table`, `view`, and `incremental` (with `on_schema_change: append_new_columns`). Views offer limited support for column names and data types, but not `constraints`. Contracts are not supported for `ephemeral`-materialized models.
-- The most commonly used [data platforms](/docs/supported-data-platforms), including Snowflake, BigQuery, Databricks, and Redshift. However, the specific `constraints` that are supported or enforced can vary depending on the platform.
+- Models materialized as `table`, `view`, and `incremental` (with `on_schema_change: append_new_columns`)
```
Comment (runleonarun) [inline website/docs/docs/collaborate/govern/model-contracts.md]:
```suggestion
- Models materialized as `table`, `view`, and `incremental` (with `on_schema_change: append_new_columns`). Views offer limited support for column names and data types, but not `constraints`. Contracts are not supported for `ephemeral`-materialized models.
```

**Comment by runleonarun** [comment]
> How do you think we should make that extensible, for all the other DWHs?
> Do you want to do that as part of the scope of this PR, or should we open a new issue?

Such good questions. I like Doug's suggestions, so we could maybe riff on them in the issue I opened to follow-up with this.  https://github.com/dbt-labs/docs.getdbt.com/issues/3282

**Comment by runleonarun** [comment]
Merging because this wording was approved by Jeremy earlier!

**Pull request merged**